runtests.py

```diff
@@ -472,5 +478,49 @@ def lcov_generate():
         print("HTML output generated under build/lcov/")
 
 
+@contextlib.contextmanager
+def working_dir(new_dir):
+    current_dir = os.getcwd()
+    try:
+        os.chdir(new_dir)
+        yield
+    finally:
+        os.chdir(current_dir)
+
+
+def run_mypy(args):
+    if args.no_build:
+        raise ValueError('Cannot run mypy with --no-build')
+
+    mypy_check = subprocess.run(
```

Comment:
Couldn't find a good way to check for NumPy stubs. Maybe I'll figure it out eventually.